app/models/ont/plate_with_samples_factory.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+# Ont namespace
+module Ont
+  # PlateWithSamplesFactory
+  # A factory for bulk inserting a plate and all of its dependents
+
+  # rubocop:disable Metrics/ClassLength
+  class PlateWithSamplesFactory
```

Comment:
It's too much faff for now, I think, but it would be nice if in the future this were split into two classes with different responsibilities.  Something like a `Serializer` which deals with serializing the hashes and then a `DatabaseInserter` that is responsible for only bulk inserting the data from those hashes.